File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        lt = 0
        rt = n-1
        max_area = 0
        while lt <= rt:
            new_area = min(height[lt], height[rt]) * (rt-lt)
            max_area = max(max_area, new_area)
            if height[lt] > height[rt]:
                rt -= 1
            else:
                lt += 1
                
        return max_area

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int lt = 0;
        int rt = height.length - 1;
        int max = 0;

        while (lt < rt) {
            max = max(min(height[lt], height[rt]) * (rt - lt), max);
            if (height[lt] > height[rt]) {
                rt--;
            }
            else {
                lt++;
            }
        }

        return max;
    }

    static int max(int a, int b) {
        return a > b ? a : b;
    }

    static int min(int a, int b) {
        return a > b ? b : a;
    }
}